src/EventGridEmulator.Tests/PullModelEventGridClientTests.cs

```diff
@@ -0,0 +1,83 @@
+using Azure.Core.Pipeline;
+using Azure;
+using EventGridEmulator.Configuration;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.TestHost;
+using Azure.Messaging;
+using Azure.Messaging.EventGrid.Namespaces;
+
+namespace EventGridEmulator.Tests;
+
+public sealed class PullModelEventGridClientTests
+{
+    // TODO Update readme
+    [Fact]
+    public async Task CanSendAndReceiveEvents()
+    {
+        var topicName = "customers";
+        var eventSubscriptionName = "CustomSubscription";
+
+        await using var factory = new CustomWebApplicationFactory(options =>
```

Comment:
Could we either split this test in function or add comments to make it easier to read (and eventually maintains)?

Reply:
We could even split in multiple test for the different scenario so if there is an error it's easier to debug.

- Publish - Receive - Acknoweldge
- Release invalide token
- Publish - Receive - Reject